src/cmd/ksh93/sh/parse.c

```diff
@@ -864,8 +865,14 @@ static_fn Shnode_t *funct(Lex_t *lexp) {
     }
     shp->st.staklist = (struct slnod *)slp;
     last = fctell();
-    fp->functline = (last - first);
-    fp->functtre = t;
+    // It should be impossible for `fp` to be NULL since a longjmp() should not occur before it has
+    // been assigned a non-NULL value in the `if (jmpval == 0) {...}` block above.
+    if (!fp) {
```

Comment:
We can avoid this `if` block by adding `assert(fp != NULL)`.

Reply:
Unfortunately cppcheck isn't smart enough to recognize that keeps the next two statements from dereferencing a NULL pointer. So rather than add two `// cppcheck-suppress` comments (one for each reference) I opted for this structure.